GAE/src/org/akvo/flow/api/app/DataPointServlet.java

```diff
@@ -111,6 +97,39 @@ protected RestResponse handleRequest(RestRequest req) throws Exception {
         return res;
     }
 
+    public List<SurveyedLocale> getDataPointList(Long surveyId, Long deviceId) {
+        List<DataPointAssignment> assList =
+                dataPointAssignmentDao.listByDeviceAndSurvey(deviceId, surveyId);
+
+
+        if (assList.isEmpty()) {
+            // Mimic old behavior
+            SurveyAssignmentDao saDao = new SurveyAssignmentDao();
+
+            List<SurveyAssignment> allAssigmentsToDevice = saDao.listAllContainingDevice(deviceId);
+            List<SurveyAssignment> filteredAssignments = allAssigmentsToDevice.stream().filter(assignment ->
+                    assignment.getSurveyId().equals(surveyId)).collect(Collectors.toList());
+
+            if (filteredAssignments.isEmpty()) {
+                return Collections.emptyList();
```

Comment:
This branch should be an exceptional one.. maybe we should log some kind of error here.  There is a survey on a device but when attempting to retrieve its assignment we find nothing